base/some.jl

```diff
@@ -0,0 +1,68 @@
+# This file is a part of Julia. License is MIT: https://julialang.org/license
+
+"""
+    Some{T}
+
+A wrapper type used with `Union{Some{T}, Void}` to distinguish between the absence
+of a value ([`nothing`](@ref)) and the presence of a `nothing` value (i.e. `Some(nothing)`).
+It can also be used to force users of a function argument or of an object field
```

Comment:
Right. I wonder whether I shouldn't remove this sentence anyway, since we don't encourage people to force manual unwrapping. APIs would be more consistent if `Some` was only used where `nothing` is a possible valid value.